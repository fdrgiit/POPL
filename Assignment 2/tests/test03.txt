s b c d e #
BEGIN END ASSIGN EQUAL DIV DO COLON ID NUM REALNUM COMMA #
s -> BEGIN b END #
b -> #
b -> ASSIGN c EQUAL d DIV d #
b -> DO c COLON e #
c -> ID #
d -> NUM #
d -> REALNUM #
e -> d #
e -> d COMMA e # 
##

