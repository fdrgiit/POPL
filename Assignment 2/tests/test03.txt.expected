FIRST(s) = { BEGIN }
FIRST(b) = { #, ASSIGN, DO }
FIRST(c) = { ID }
FIRST(d) = { NUM, REALNUM }
FIRST(e) = { NUM, REALNUM }
FOLLOW(s) = { $ }
FOLLOW(b) = { END }
FOLLOW(c) = { COLON, EQUAL }
FOLLOW(d) = { COMMA, DIV, END }
FOLLOW(e) = { END }
